### DISCUSSION TITLE
Test file-uploading API Endpoint

### DISCUSSION THREAD

**Discussion by machineghost** [Xunnamius/next-test-api-route-handler#837] [Q&A] [answered]
I just discovered this library, and it seems to be exactly what I need, except ... I need to test a file-upload API, and I couldn't find any examples in the documentation of testing such endpoints.

Is it possible, and if so are there any examples or docs I can reference?

**Comment by Xunnamius** [comment, marked as answer]
Hi there. Each invocation of [`testApiHandler`](https://github.com/Xunnamius/next-test-api-route-handler#usage) spins up a tiny transient HTTP server on a random port, to which the [`fetch` function](https://github.com/Xunnamius/next-test-api-route-handler#test) sends your requests. That server calls out to the Next.js internal function `apiResolver`, which handles the request entirely.

If you want to test that your endpoint handles file uploads as expected, it should be no different than uploading a file using `fetch` [normally](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_a_file).
